### PR TITLE
Replaced org.jitsi.util.StringUtils to org.apache.commons.lang3.StringUtils

### PR DESCRIPTION
**Pull request by GNUDimarik**

### CODE REVIEW

**Comment by GNUDimarik** [comment]
@ibauersachs sorry if I distract you to this mesage. But please take a look once you have time on it

File: src/net/java/sip/communicator/plugin/jabberaccregwizz/IceConfigPanel.java [hunk 1]
```diff
@@ -202,14 +205,25 @@ public void actionPerformed(ActionEvent e)
 
                 if (stunServer != null)
                 {
+                    String userName;
+                    String password;
+
+                    try {
```
Comment (ibauersachs) [inline src/net/java/sip/communicator/plugin/jabberaccregwizz/IceConfigPanel.java]: Formatting

File: src/net/java/sip/communicator/plugin/jabberaccregwizz/IceConfigPanel.java [hunk 2]
```diff
@@ -202,14 +205,25 @@ public void actionPerformed(ActionEvent e)
...
+                        userName = StringUtils.toString(
```
Comment (ibauersachs) [inline src/net/java/sip/communicator/plugin/jabberaccregwizz/IceConfigPanel.java]: This is a deprecated method. Use new String(..., StandardCharets.UTF_8) or if it throws use toEncodedString

File: src/net/java/sip/communicator/plugin/jabberaccregwizz/IceConfigPanel.java [hunk 3]
```diff
@@ -202,14 +205,25 @@ public void actionPerformed(ActionEvent e)
...
+                            stunServer.getUsername(), StandardCharsets.UTF_8.name());
+                        password = StringUtils.toString(
+                            stunServer.getPassword(), StandardCharsets.UTF_8.name());
+                    } catch (UnsupportedEncodingException ex) {
```
Comment (ibauersachs) [inline src/net/java/sip/communicator/plugin/jabberaccregwizz/IceConfigPanel.java]: More formatting

File: src/net/java/sip/communicator/plugin/loggingutils/LoggingConfigForm.java [hunk 4]
```diff
@@ -246,7 +247,7 @@ private void init()
         c.gridy = 4;
         mainPanel.add(archiveButton, c);
 
-        if(!StringUtils.isNullOrEmpty(getUploadLocation()))
+        if(StringUtils.isEmpty(getUploadLocation()))
```
Comment (ibauersachs) [inline src/net/java/sip/communicator/plugin/loggingutils/LoggingConfigForm.java]: isNotEmpty

File: src/net/java/sip/communicator/service/protocol/StunServerDescriptor.java [hunk 5]
```diff
@@ -248,8 +250,16 @@ public void storeDescriptor(Map<String, String> props, String namePrefix)
             props.put(namePrefix + STUN_PORT, Integer.toString( getPort() ));
 
         if (getUsername() != null && getUsername().length > 0)
-            props.put(namePrefix + STUN_USERNAME,
-                      StringUtils.getUTF8String(getUsername()));
+            try
+            {
+                props.put(namePrefix + STUN_USERNAME,
+                          StringUtils.toString(getUsername(), StandardCharsets.UTF_8.name()));
```
Comment (ibauersachs) [inline src/net/java/sip/communicator/service/protocol/StunServerDescriptor.java]: use new String(..., StandardCharsets.UTF_8). Or if this throws, use toEncodedString

File: src/net/java/sip/communicator/plugin/desktoputil/plaf/SearchFieldUI.java [hunk 6]
```diff
@@ -177,10 +179,12 @@ protected void customPaintBackground(Graphics g)
                 - searchIcon.getIconHeight()/2;
 
             g2.drawImage(searchIcon.getImage(), 3, dy, null);
+            Pattern p = Pattern.compile(".*[a-zA-Z].*");
+            Matcher m = p.matcher(c.getText());
 
             if (c.getText() != null
```
Comment (ibauersachs) [inline src/net/java/sip/communicator/plugin/desktoputil/plaf/SearchFieldUI.java]: Use isNotEmpty(c.getText()) && !StringUtils.isAlpha(...)

Not sure what the test actually wants to check. Numbers only? Then the regex should be "\\d+". If it's meant to be a phone number, what about spaces, + for international, etc.

File: src/net/java/sip/communicator/impl/gui/main/contactlist/UnknownContactPanel.java [hunk 7]
```diff
@@ -190,9 +192,11 @@ public void actionPerformed(ActionEvent e)
      */
     private void initSMSButton()
     {
+        Pattern p = Pattern.compile(".*[a-zA-Z].*");
+        Matcher m = p.matcher(parentWindow.getCurrentSearchText());
+
         if(parentWindow.hasOperationSet(OperationSetSmsMessaging.class)
```
Comment (ibauersachs) [inline src/net/java/sip/communicator/impl/gui/main/contactlist/UnknownContactPanel.java]: isAlpha

**Comment by GNUDimarik** [comment]
@ibauersachs thanks for review!
Will update this PR soon

File: src/net/java/sip/communicator/plugin/loggingutils/LoggingConfigForm.java (hunk 4 above)
Comment (GNUDimarik) [reply src/net/java/sip/communicator/plugin/loggingutils/LoggingConfigForm.java]: Thanks! I tried to review it by my self before commiting for catching the similar issues. But seems I miss one :-)

File: src/net/java/sip/communicator/plugin/desktoputil/plaf/SearchFieldUI.java (hunk 6 above)
Comment (GNUDimarik) [reply src/net/java/sip/communicator/plugin/desktoputil/plaf/SearchFieldUI.java]: By old code check wanted to know is there letters in the string or not

Comment (GNUDimarik) [reply src/net/java/sip/communicator/plugin/desktoputil/plaf/SearchFieldUI.java]: May be [isAlphanumericSpace](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#isAlphanumericSpace-java.lang.CharSequence-)?

**Comment by GNUDimarik** [comment]
@ibauersachs please check second commit once you have time

**Comment by GNUDimarik** [comment]
@ibauersachs sorry. Didn't check manifests. Will do.

**Comment by GNUDimarik** [comment]
@ibauersachs @damencho Added imports to manifests. But no deep QA ...
Please check once you able

File: src/net/java/sip/communicator/impl/gui/main/contactlist/UnknownContactPanel.java [hunk 8]
```diff
@@ -20,6 +20,8 @@
 import java.awt.*;
 import java.awt.event.*;
 import java.util.List;
+import java.util.regex.Matcher;
```
Comment (ibauersachs) [inline src/net/java/sip/communicator/impl/gui/main/contactlist/UnknownContactPanel.java]: These are now unnecessary again

File: src/net/java/sip/communicator/plugin/desktoputil/plaf/SearchFieldUI.java [hunk 9]
```diff
@@ -177,10 +179,21 @@ protected void customPaintBackground(Graphics g)
                 - searchIcon.getIconHeight()/2;
 
             g2.drawImage(searchIcon.getImage(), 3, dy, null);
+            String str = c.getText();
+            boolean containsLetter = false;
+
+            for (int i = 0; i < str.length(); i++)
+            {
+                if (Character.isLetter(str.charAt(i)))
```
Comment (ibauersachs) [inline src/net/java/sip/communicator/plugin/desktoputil/plaf/SearchFieldUI.java]: Possible NPE. Please, THINK and ANALYZE before doing changes.

Use isAlpha here as well. I doube the code is used at all anyway.

File: src/net/java/sip/communicator/plugin/jabberaccregwizz/IceConfigPanel.java [hunk 10]
```diff
@@ -202,14 +205,27 @@ public void actionPerformed(ActionEvent e)
...
+                    String password;
+
+                    try
+                    {
+                        userName = new String(
+                            stunServer.getUsername(), StandardCharsets.UTF_8.name());
+                        password = new String(
+                            stunServer.getPassword(), StandardCharsets.UTF_8.name());
+                    }
+                    catch (UnsupportedEncodingException ex) {
```
Comment (ibauersachs) [inline src/net/java/sip/communicator/plugin/jabberaccregwizz/IceConfigPanel.java]: Formatting.
As I said, 'if it throws, use StringUtils.toEncodedString', so why don't you?

File: src/net/java/sip/communicator/service/protocol/StunServerDescriptor.java [hunk 11]
```diff
@@ -248,8 +250,16 @@ public void storeDescriptor(Map<String, String> props, String namePrefix)
             props.put(namePrefix + STUN_PORT, Integer.toString( getPort() ));
 
         if (getUsername() != null && getUsername().length > 0)
-            props.put(namePrefix + STUN_USERNAME,
-                      StringUtils.getUTF8String(getUsername()));
+            try
+            {
+                props.put(namePrefix + STUN_USERNAME,
```
Comment (ibauersachs) [inline src/net/java/sip/communicator/service/protocol/StunServerDescriptor.java]: toEncodedString

File: src/net/java/sip/communicator/plugin/desktoputil/plaf/SearchFieldUI.java (hunk 9 above)
Comment (GNUDimarik) [reply src/net/java/sip/communicator/plugin/desktoputil/plaf/SearchFieldUI.java]: Right. Sorry. No rush anymore. FIxed

File: src/net/java/sip/communicator/service/protocol/StunServerDescriptor.java (hunk 11 above)
Comment (GNUDimarik) [reply src/net/java/sip/communicator/service/protocol/StunServerDescriptor.java]: It could not to resolve symbol for some reason.
Here is the output:
    [javac] D:\Source\Repos\jitsi\src\net\java\sip\communicator\plugin\jabberaccregwizz\IceConfigPanel.java:213: error: cannot find symbol
    [javac]                         userName = StringUtils.toEncodedString(stunServer.getUsername(), StandardCharsets.UTF_8);
    [javac]                                               ^
    [javac]   symbol:   method toEncodedString(byte[],Charset)

File: src/net/java/sip/communicator/plugin/jabberaccregwizz/IceConfigPanel.java (hunk 10 above)
Comment (GNUDimarik) [reply src/net/java/sip/communicator/plugin/jabberaccregwizz/IceConfigPanel.java]: Could not resolve symbol

**Comment by GNUDimarik** [comment]
@ibauersachs Please check once more. Fixed comments except ones with request of using StringUtils.eotEncodedString. It could not to resolve it for some reason. Please guide me a little

**Comment by GNUDimarik** [comment]
@ibauersachs I see there are 2 unfixed comments .... Going to investigate them soon. 
Thanks for review!

**Comment by GNUDimarik** [comment]
@ibauersachs toEncodedString issue fixed after merging latest upstream. Please check. I scare to miss something since I'm exhousted now ...

**Comment by GNUDimarik** [comment]
@ibauersachs please let me know if I need to change something

File: src/net/java/sip/communicator/service/protocol/StunServerDescriptor.java (hunk 11 above)
Comment (ibauersachs) [reply src/net/java/sip/communicator/service/protocol/StunServerDescriptor.java]: Please get rid of the try/catch

File: src/net/java/sip/communicator/service/protocol/media/CallPeerMediaHandler.java [hunk 12]
```diff
@@ -1595,15 +1595,16 @@ protected boolean requestKeyFrame()
      * @param stream <tt>MediaStream</tt> non-null stream
      * @param mediaType <tt>MediaType</tt>
      */
-    protected void sendHolePunchPacket(MediaStream stream, MediaType mediaType)
+    public void sendHolePunchPacket(MediaStream stream, MediaType mediaType)
```
Comment (ibauersachs) [inline src/net/java/sip/communicator/service/protocol/media/CallPeerMediaHandler.java]: why?

File: src/net/java/sip/communicator/service/protocol/media/CallPeerMediaHandler.java [hunk 13]
```diff
@@ -1595,15 +1595,16 @@ protected boolean requestKeyFrame()
...
         if (disableHolePunching)
             return;
 
         // send as a hole punch packet a constructed rtp packet
         // has the correct payload type and ssrc
-        RawPacket packet = new RawPacket(
-            HOLE_PUNCH_PACKET, 0, RawPacket.FIXED_HEADER_SIZE);
+        RawPacket packet
+            = new RawPacket(
+                    HOLE_PUNCH_PACKET.clone(), 0, HOLE_PUNCH_PACKET.length);
```
Comment (ibauersachs) [inline src/net/java/sip/communicator/service/protocol/media/CallPeerMediaHandler.java]: why?

File: src/net/java/sip/communicator/service/protocol/media/CallPeerMediaHandler.java (hunk 12 above)
Comment (GNUDimarik) [reply src/net/java/sip/communicator/service/protocol/media/CallPeerMediaHandler.java]: Really don't remember this edit!!!

Comment (GNUDimarik) [reply src/net/java/sip/communicator/service/protocol/media/CallPeerMediaHandler.java]: ah @ibauersachs I merged upstream to my origin. Seems soething was wrong. WIll checj

Comment (GNUDimarik) [reply src/net/java/sip/communicator/service/protocol/media/CallPeerMediaHandler.java]: @ibauersachs this is the master
https://github.com/jitsi/jitsi/blob/master/src/net/java/sip/communicator/service/protocol/media/CallPeerMediaHandler.java#L1598
So the issue is: this pull request contains merge commits.

Comment (GNUDimarik) [reply src/net/java/sip/communicator/service/protocol/media/CallPeerMediaHandler.java]: I'll close it and create new one with one huge commit

Comment (ibauersachs) [reply src/net/java/sip/communicator/service/protocol/media/CallPeerMediaHandler.java]: No, please don't. I can't review 100+ files again.

Comment (ibauersachs) [reply src/net/java/sip/communicator/service/protocol/media/CallPeerMediaHandler.java]: > @ibauersachs this is the master
> https://github.com/jitsi/jitsi/blob/master/src/net/java/sip/communicator/service/protocol/media/CallPeerMediaHandler.java#L1598
> So the issue is: this pull request contains merge commits.

I don't know why this shows up in the diff then, but okay, doesn't matter then.

Comment (GNUDimarik) [reply src/net/java/sip/communicator/service/protocol/media/CallPeerMediaHandler.java]: @ibauersachs please check once you have time

File: src/net/java/sip/communicator/plugin/jabberaccregwizz/IceConfigPanel.java [hunk 14]
```diff
@@ -19,6 +19,8 @@
 
 import java.awt.*;
 import java.awt.event.*;
+import java.io.UnsupportedEncodingException;
```
Comment (ibauersachs) [inline src/net/java/sip/communicator/plugin/jabberaccregwizz/IceConfigPanel.java]: Where is this used?

File: src/net/java/sip/communicator/plugin/desktoputil/plaf/SearchFieldUI.java [hunk 15]
```diff
@@ -177,10 +180,25 @@ protected void customPaintBackground(Graphics g)
                 - searchIcon.getIconHeight()/2;
 
             g2.drawImage(searchIcon.getImage(), 3, dy, null);
+            String str = c.getText();
```
Comment (ibauersachs) [inline src/net/java/sip/communicator/plugin/desktoputil/plaf/SearchFieldUI.java]: Either:
- Keep using `StringUtils.containsLetters` (from Jitsi-Utils) and undo the changes here
- Or simply test for !isAlpha
Just copy/pasting a method from jitsi-utils to here is besides the point.

Please also fix the formatting and don't double check for null/empty (currently lines 186 and 199/200).

Comment (GNUDimarik) [reply src/net/java/sip/communicator/plugin/desktoputil/plaf/SearchFieldUI.java]: Copy pasting methods allows us to remove dependency fully and in the same time save the same mention as previously.
But you are the boss ...

Comment (GNUDimarik) [reply src/net/java/sip/communicator/plugin/desktoputil/plaf/SearchFieldUI.java]: But yes. I have better solution

**Comment by GNUDimarik** [comment]
@ibauersachs please review once you have time

**Comment by GNUDimarik** [comment]
@ibauersachs any update?
